front/pages/w/[wId]/subscription/awaiting_confirmation.tsx

```diff
@@ -0,0 +1,93 @@
+import { BarHeader, Page, Spinner2 } from "@dust-tt/sparkle";
+import type { UserType, WorkspaceType } from "@dust-tt/types";
+import type { SubscriptionType } from "@dust-tt/types";
+import type { InferGetServerSidePropsType } from "next";
+import { useRouter } from "next/router";
+import React, { useEffect, useState } from "react";
+
+import config from "@app/lib/api/config";
+import { withDefaultUserAuthRequirements } from "@app/lib/iam/session";
+import { getStripeSubscription } from "@app/lib/plans/stripe";
+
+export const getServerSideProps = withDefaultUserAuthRequirements<{
+  owner: WorkspaceType;
+  subscription: SubscriptionType;
+  user: UserType;
+}>(async (context, auth) => {
+  const owner = auth.workspace();
+  const subscription = auth.subscription();
+  const user = auth.user();
+  if (!owner || !auth.isAdmin() || !subscription || !user) {
+    return {
+      notFound: true,
+    };
+  }
+
+  if (subscription.stripeSubscriptionId) {
+    const stripeSubscription = await getStripeSubscription(
+      subscription.stripeSubscriptionId
+    );
+    if (!stripeSubscription) {
+      return {
+        notFound: true,
+      };
+    }
+    stripeSubscription;
+  }
+
+  return {
+    props: {
+      owner,
+      subscription,
+      gaTrackingId: config.getGaTrackingId(),
+      user,
+    },
+  };
+});
+
+export default function Subscription({
+  owner,
+  subscription,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const router = useRouter();
+
+  useState(false);
+  useEffect(() => {
+    if (router.query.type === "succeeded") {
+      if (subscription.plan.code === router.query.plan_code) {
+        // Then we remove the query params to avoid going through this logic again.
+        void router.replace(
+          { pathname: `/w/${owner.sId}/congratulations` },
+          undefined,
+          {
+            shallow: true,
```

Comment:
You might want to remove this `shallow` here as it will prevent running data fetching methods (like `getServerSideProps`).